ipytone/ipytone.py

```diff
@@ -3,18 +3,155 @@
 
 # Copyright (c) Benoit Bovy.
 # Distributed under the terms of the Modified BSD License.
+import re
+
+from ipywidgets import widget_serialization, Widget
+from traitlets import Enum, Instance, Unicode, List, Float, Int, Bool, validate, TraitError
 
-from ipywidgets import Widget
-from traitlets import Unicode, Float, Int, Bool, validate, TraitError
 from ._frontend import module_name, module_version
 
 
-class Oscillator(Widget):
+class ToneWidgetBase(Widget):
+    _model_module = Unicode(module_name).tag(sync=True)
+    _model_module_version = Unicode(module_version).tag(sync=True)
+
+
+class AudioNode(ToneWidgetBase):
```

Comment:
I think it will be useful to give a name to nodes, so that we get a nice representation when we show the graph.